packages/desktop-client/src/components/ManageRules.tsx

```diff
@@ -97,23 +97,23 @@ function ruleToString(rule, data) {
   );
 }
 
-type ManageRulesContentProps = {
+type ManageRulesProps = {
   isModal: boolean;
   payeeId: string | null;
   setLoading?: Dispatch<SetStateAction<boolean>>;
 };
 
-function ManageRulesContent({
+export function ManageRules({
   isModal,
   payeeId,
-  setLoading,
-}: ManageRulesContentProps) {
+  setLoading = () => {},
+}: ManageRulesProps) {
   const [allRules, setAllRules] = useState([]);
   const [page, setPage] = useState(0);
   const [filter, setFilter] = useState('');
   const dispatch = useDispatch();
 
-  const { data: schedules } = SchedulesQuery.useQuery();
+  const { data: schedules = [] } = useSchedules();
```

Comment:
This is the crux of the issue I believe.

The e2e test was trying to filter the schedules list BEFORE the schedules were loaded. So the `schedules` variable would be `undefined`. And this would blow up later in the call tree.

So I patched this by moving it to `useSchedules` (which uses `useQuery`). And also by adding a default value - empty array.

Every other change is just cleaning up stuff.. removing the usages of `SchedulesQuery` and all the code that this stuff used.